spiffe-helper.go

```
@@ -96,72 +124,83 @@ func (s *Sidecar) signalProcess() (err error) {
-			Type:  "CERTIFICATE",
-			Bytes: bundle.Svid})
+	err = s.writeKey(svidKeyFile, svid.X509SvidKey)
+	if err != nil {
+		return err
+	}
 
-	log("Writing: %v\n", svidKeyFile)
-	err = ioutil.WriteFile(svidKeyFile, append(svidPrivateKey, svid...), os.ModePerm)
+	err = s.writeCerts(svidBundleFile, svid.Bundle)
 	if err != nil {
-		return ttl, fmt.Errorf("error writing file: %v\n%v", svidKeyFile, err)
+		return err
 	}
 
-	log("Writing: %v\n", svidFile)
-	err = ioutil.WriteFile(svidFile, svid, os.ModePerm)
+	return nil
+}
+
+// writeCerts takes a slice of bytes, which may contain multiple certificates,
+// and encodes them as PEM blocks, writing them to file
+func (s *sidecar) writeCerts(file string, data []byte) error {
+	certs, err := x509.ParseCertificates(data)
 	if err != nil {
-		return ttl, fmt.Errorf("error writing file: %v\n%v", svidFile, err)
+		return err
 	}
 
-	svidBundle := pem.EncodeToMemory(
-		&pem.Block{
+	pemData := []byte{}
+	for _, cert := range certs {
+		b := &pem.Block{
 			Type:  "CERTIFICATE",
-			Bytes: bundle.SvidBundle,
-		})
+			Bytes: cert.Raw,
+		}
+		pemData = append(pemData, pem.EncodeToMemory(b)...)
+	}
 
-	log("Writing: %v\n", svidBundleFile)
-	err = ioutil.WriteFile(svidBundleFile, svidBundle, os.ModePerm)
-	if err != nil {
-		return ttl, fmt.Errorf("error writing file: %v\n%v", svidBundleFile, err)
+	return ioutil.WriteFile(file, pemData, os.ModePerm)
+}
+
+// writeKey takes a private key as a slice of bytes,
+// formats as PEM, and writes it to file
+func (s *sidecar) writeKey(file string, data []byte) error {
+	b := &pem.Block{
+		Type:  "EC PRIVATE KEY",
+		Bytes: data,
 	}
 
-	return ttl, nil
+	return ioutil.WriteFile(file, pem.EncodeToMemory(b), os.ModePerm)
+}
+
+// parses a time.Duration from the the SidecarConfig,
+// if there's an error during parsing, maybe because
+// it's not well defined or not defined at all in the
+// config, returns the default timeout constant
+func getTimeout(config *SidecarConfig) time.Duration {
```

seems more resilient to fail if the duration is set but invalid.

---

do you mean that getTimeout should return an error if the duration in the conf is invalid ?